### PR TITLE
Enable doxygen for bits/module.h and document APIs using doxygen tags

### PR DESCRIPTION
**Pull request by radhikaj**
Issue #1961

### CODE REVIEW

**Comment by andschwa** [comment]
Wrong PR!

File: include/openenclave/bits/module.h [hunk 1]
```diff
@@ -19,19 +25,65 @@
...
-/** Load the Intel protected file system module. */
+/**
+ * Load the sgx file sytstem module.
+ *
+ * This function enables the use of the SGX file system module.
+ *
+ * @retval OE_OK The module was successfully loaded.
+ * @retval OE_FAILURE Module failed to load.
+ *
+ */
```
Comment (radhikaj) [inline include/openenclave/bits/module.h]: oe_load_module_sgx_file_system and oe_load_module_host_epoll could be removed as tracked in https://github.com/openenclave/openenclave/issues/1917. But until then, these can be documented in gh pages.

**Comment by andschwa** [comment]
Flaky Windows...

bors r+

**Pull request merged**